### ISSUE TITLE
Upgrade kernel to 4.20.9 to fix nfs-client bug

### ISSUE THREAD

**Issue by ruslanloman** [flatcar/Flatcar#80]
**Description**                                                                          
We faced with nfs client bug related to NFSv4 delegation feature as result nfs client just stuck and only node reboot helps to solve issue, it was fixed in  4.20 kernel version  [Fix lock recovery during NFSv4 delegation recalls](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=c7a2c49ea6c9eebbe44ff2c08b663b2905ee2c13). Disabling nfs delegation feature solve issue, but in that case load on nfs-server is increased twice.

**Environment and steps to reproduce**
Unfortunately there is no strict rule how to reproduce it. It happens on our production environment multiple times per day.

In nfs client log we see that it's just stuck in loop by checking same test_stateid 00000000ec5d02eb:
```
Thu Apr  9 15:18:57 2020] NFS reply test_stateid: succeeded, 0
[Thu Apr  9 15:18:57 2020] NFS call  test_stateid 00000000ec5d02eb
[Thu Apr  9 15:18:57 2020] --> nfs41_call_sync_prepare data->seq_server 000000006dfc86c9
[Thu Apr  9 15:18:57 2020] --> nfs4_alloc_slot used_slots=0000 highest_used=4294967295 max_slots=31
[Thu Apr  9 15:18:57 2020] <-- nfs4_alloc_slot used_slots=0001 highest_used=0 slotid=0
[Thu Apr  9 15:18:57 2020] encode_sequence: sessionid=1585584999:2538115180:5741:0 seqid=13899231 slotid=0 max_slotid=0 cache_this=0
[Thu Apr  9 15:18:57 2020] nfs41_handle_sequence_flag_errors: "10.1.4.65" (client ID 671b825e6c904897) flags=0x00000040
[Thu Apr  9 15:18:57 2020] --> nfs4_alloc_slot used_slots=0001 highest_used=0 max_slots=31
[Thu Apr  9 15:18:57 2020] <-- nfs4_alloc_slot used_slots=0003 highest_used=1 slotid=1
[Thu Apr  9 15:18:57 2020] nfs4_free_slot: slotid 1 highest_used_slotid 0
[Thu Apr  9 15:18:57 2020] nfs41_sequence_process: Error 0 free the slot 
[Thu Apr  9 15:18:57 2020] nfs4_free_slot: slotid 0 highest_used_slotid 4294967295
[Thu Apr  9 15:18:57 2020] NFS reply test_stateid: succeeded, 0
[Thu Apr  9 15:18:57 2020] NFS call  test_stateid 00000000ec5d02eb
[Thu Apr  9 15:18:57 2020] --> nfs41_call_sync_prepare data->seq_server 000000006dfc86c9
[Thu Apr  9 15:18:57 2020] --> nfs4_alloc_slot used_slots=0000 highest_used=4294967295 max_slots=31
[Thu Apr  9 15:18:57 2020] <-- nfs4_alloc_slot used_slots=0001 highest_used=0 slotid=0
[Thu Apr  9 15:18:57 2020] encode_sequence: sessionid=1585584999:2538115180:5741:0 seqid=13899232 slotid=0 max_slotid=0 cache_this=0
[Thu Apr  9 15:18:57 2020] nfs41_handle_sequence_flag_errors: "10.1.4.65" (client ID 671b825e6c904897) flags=0x00000040
[Thu Apr  9 15:18:57 2020] --> nfs4_alloc_slot used_slots=0001 highest_used=0 max_slots=31
[Thu Apr  9 15:18:57 2020] <-- nfs4_alloc_slot used_slots=0003 highest_used=1 slotid=1
[Thu Apr  9 15:18:57 2020] nfs4_free_slot: slotid 1 highest_used_slotid 0
[Thu Apr  9 15:18:57 2020] nfs41_sequence_process: Error 0 free the slot 
[Thu Apr  9 15:18:57 2020] nfs4_free_slot: slotid 0 highest_used_slotid 4294967295
[Thu Apr  9 15:18:57 2020] NFS reply test_stateid: succeeded, 0
[Thu Apr  9 15:18:57 2020] NFS call  test_stateid 00000000ec5d02eb
[Thu Apr  9 15:18:57 2020] --> nfs41_call_sync_prepare data->seq_server 000000006dfc86c9
[Thu Apr  9 15:18:57 2020] --> nfs4_alloc_slot used_slots=0000 highest_used=4294967295 max_slots=31
[Thu Apr  9 15:18:57 2020] <-- nfs4_alloc_slot used_slots=0001 highest_used=0 slotid=0
[Thu Apr  9 15:18:57 2020] encode_sequence: sessionid=1585584999:2538115180:5741:0 seqid=13899233 slotid=0 max_slotid=0 cache_this=0
[Thu Apr  9 15:18:57 2020] nfs41_handle_sequence_flag_errors: "10.1.4.65" (client ID 671b825e6c904897) flags=0x00000040
[Thu Apr  9 15:18:57 2020] --> nfs4_alloc_slot used_slots=0001 highest_used=0 max_slots=31
[Thu Apr  9 15:18:57 2020] <-- nfs4_alloc_slot used_slots=0003 highest_used=1 slotid=1
[Thu Apr  9 15:18:57 2020] nfs4_free_slot: slotid 1 highest_used_slotid 0
[Thu Apr  9 15:18:57 2020] nfs41_sequence_process: Error 0 free the slot 
[Thu Apr  9 15:18:57 2020] nfs4_free_slot: slotid 0 highest_used_slotid 4294967295
```

**Impact**
This is impacting all users who use nfs 4.x delegation feature.

**Additional information**
flatcar release version: 2345.3.1
nfs mount options: 
`rw,nosuid,nodev,noexec,noatime,vers=4.2,rsize=1048576,wsize=1048576,namlen=255,acregmin=600,acregmax=600,acdirmin=600,acdirmax=600,soft,proto=tcp,timeo=10,retrans=2,sec=sys,local_lock=none"`

**Comment by dongsupark** [comment]
Thanks for the report.

As far as I understand, you are saying that the NFS lock recovery issue was fixed via [44f411c353bf](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=44f411c353bf6d98d5a34f8f1b8605d43b2e50b8) ("NFSv4.x: fix lock recovery during delegation recall") . True, it was first included in Kernel 4.20.

However, Flatcar stable 2345.3.1 has Kernel 4.19.107, which already includes its backported commit [90d73c1cadb8](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?id=90d73c1cadb8af69fb8805638c29a9af482b46a9). The commit has been available since v4.19.86.

So I suppose, the bug could be fixed in v4.19.86, but started to appear again at some point between 4.19.86 and 4.19.107. Or it could be related to other reasons.

**Comment by ruslanloman** [comment]
Hi,
Thanks for pointing to backported commit. We proceed debug issue to collect more information, but i think we can close this issue since it's not relevant anymore.
Thanks!
